05_iigo/index.tex

```diff
@@ -500,4 +500,226 @@ \subsection{Legislative Branch}
+If our goal is simply to ensure that an agent pays the expected amount of tax, a basic mathematical operation we can perform is subtract the actual tax paid from the expected.
+If we get 0, we know this agent has adhered to this rule. 
+\[
+  \textrm{let expected tax} = x   
+\]
+\[
+    \textrm{let actual paid tax} = y    
+\]
+\[
+    \textrm{We want    }
+      y - x = 0
+\]
+To turn this into a matrix calculation, we simply have to introduce some trivial coefficients.
+\[
+      -1*x +
+       1*y = 0
+\]
+Now we can write this as a matrix calculation.
+\[
+    \begin{bmatrix}
+        -1 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y
+    \end{bmatrix}   
+    = 0  
+\]
+
+We have now encoded this simply rule as a matrix product. An agent can input their versions of x and y, and perform the calculation.
+The agent can then check if their result is 0, if so they are compliant and can pay $y$ tax. Let us revisit our rule, and evolve it a little.
+\linebreak
+\linebreak
+"If you are expected to pay x amount of tax, the amount of tax you pay must be at least x"
+\linebreak
+\linebreak
+We can rewrite the matrix equation as an inequality.
+\[
+    \begin{bmatrix}
+        -1 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y
+    \end{bmatrix}   
+    >= 0  
+\]
+The addition of the "at least" adds a little bit of complexity to our matrix schema. In that we may not be able to simply check for equality to a constant, and may instead need to check for a certain condition.
+To make these comparisons easier, we have decided to ensure that whatever condition, equality or inequality, we would compare to 0.
+However if our tax rule looked like this (for whatever reason):
+\linebreak
+\linebreak
+"If you are expected to pay x amount of tax, the amount of tax you pay must be at least x + 5"
+\linebreak
+\linebreak
+Then it would be impossible to capture that using the current matrix calculation, so to capture these cases we introduce a constant to the input list, as below.
+\[
+    \begin{bmatrix}
+        -1 & 1 & -5
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y \\
+        1
+    \end{bmatrix}   
+    >= 0  
+\]
+We have now managed to capture any linear equation or inequality with any constant shift. This is where we decided to limit the scope of our rules from a technical aspect.
+The final section of our schema is how to inform an agent whether it should be looking for equality or inequality.
+For this, we package with the rule-matrix an "auxiliary" vector containing a list of comparison codes.
+\begin{table}[h]
+    \centering
+    \begin{tabular}{ll}
+    \hline
+    \multicolumn{1}{|l}{Aux code} & \multicolumn{1}{l|}{Meaning} \\ \hline
+    0                             & = 0                          \\
+    1                             & \textgreater 0               \\
```

Comment:
$\geq$ for greater than equal....

similarly, could you PLZ find the relevant latex commands..
<= does not look very good lol